### ISSUE TITLE
Tuple not displaying properly in REPL

### ISSUE THREAD

**Issue by wheineman** [JuliaLang/julia#7434]
````
$ julia
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "help()" to list help topics
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.3.0-prerelease+3908 (2014-06-26 23:41 UTC)
 _/ |\__'_|_|_|\__'_|  |  Commit 8f99257 (0 days old master)
|__/                   |  x86_64-w64-mingw32

julia> x = tuple(1,2,3)
(1,2,0)

julia> x
(1,2,0)

julia> x[3]
3

julia>
julia> versioninfo()
Julia Version 0.3.0-prerelease+3908
Commit 8f99257 (2014-06-26 23:41 UTC)
Platform Info:
  System: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i7-4600M CPU @ 2.90GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY)
  LAPACK: libopenblas
  LIBM: libopenlibm

julia>

**Comment by IainNZ** [comment]
Another case on SO: http://stackoverflow.com/questions/28423504/why-does-julia-1-2-3-return-1-2-0/28423717#28423717

**Comment by adijkstra** [comment]
I also still have this issue in 0.3.5

julia> A = (1,2,3)
(1,2,0)

julia> A
(1,2,0)

julia> versioninfo()
Julia Version 0.3.5
Commit a05f87b* (2015-01-08 22:33 UTC)
Platform Info:
  System: Windows (x86_64-w64-mingw32)
  CPU: Intel(R) Core(TM) i3-3220 CPU @ 3.30GHz
  WORD_SIZE: 64
  BLAS: libopenblas (USE64BITINT DYNAMIC_ARCH NO_AFFINITY Sandybridge)
  LAPACK: libopenblas
  LIBM: libopenlibm
  LLVM: libLLVM-3.3

**Comment by pao** [comment]
@adijkstra, that is expected; we are unlikely to be updating the LLVM dependency to 3.5 or later, where the problem is fixed, in the 0.3.x series of Julia releases due to extensive changes in LLVM's APIs. Note from your `versioninfo()` output that your build is using LLVM 3.3.

**Comment by tkelman** [comment]
Given that it's looking like we'll also have to stay on LLVM 3.3 for Julia 0.4.0, would it be worth digging into exactly where in the show pipeline this happens? I unfortunately don't have reliable access to a machine where this happens.
